contracts/token/BaseTokenSweep.sol

```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+
+/**
+ * @title
+ *
+ * @dev
+ */
+abstract contract BaseTokenSweep is UUPSUpgradeable {
+    address public beneficiary;
+
+    event BeneficiaryUpdated(address indexed beneficiary);
+
+    /**
+     * @notice Set the beneficiary of the token sweep
+     *
+     * @dev Needs access control implemented in the inheriting contract
+     * @param _beneficiary The address of the beneficiary
+     */
+    function setBeneficiary(address _beneficiary) external virtual {
+        require(_beneficiary != address(0), "WithdrawTokens/beneficiary-zero");
+        require(_beneficiary != address(this), "WithdrawToken/self-address");
+
+        beneficiary = _beneficiary;
+        emit BeneficiaryUpdated(_beneficiary);
+    }
+
+    /**
+     * @notice An initializer instead of a constructor.
+     *
+     * @dev Compared to a constructor, an init adds deployment cost (as constructor code is executed but not deployed).
+     *      However when used in conjunction with a proxy, the init means the contract can be upgraded.
+     */
+    function initialize() external virtual;
```

Comment:
nit; keep the init name inline with OZ `BaseTokenSweep__init()`.

This approach makes more sense when you see contract that implement a bunch of different abstracts, each with their own init.

Reply:
Do we still want this initialize() function?

> In the upgradable contract I've been writing, there's only been a need for _authorizeUpgrade and that's to add permission modifiers.

I've removed it as per this comment above

Good to know about the initialiser naming convention - i often wondered why OZ had such long names!

Reply:
Yeah, ended up having to disable one of the Slither detectors because if it 😞

Reply:
Drop the `initialize()` function. Typically I'm seeing that is used only in the concrete contracts (that are not expected to be extended).